src/conspiracies/preprocessing/create_context_threads.py

```diff
@@ -135,7 +135,7 @@ def range_of_dates(start_date, extra_days):
     # write contests to ndjson; the ndjson of contexts has the format: List[Dict[str, Any]]
     with open(
         os.path.join(
-            f"tweet_threads_{datetime.datetime.strptime(args.start_date, '%Y-%m-%d').date() }_{datetime.datetime.strptime(args.start_date, '%Y-%m-%d').date() + datetime.timedelta(days=args.extra_days-1)}.ndjson",
+            f"tweet_threads_{datetime.datetime.strptime(args.start_date, '%Y-%m-%d').date()}_{datetime.datetime.strptime(args.start_date, '%Y-%m-%d').date() + datetime.timedelta(days=args.extra_days-1)}.ndjson",
         ),
         "w",
     ) as f:
```

Comment:
if you use pathlib here as well it becomes:

```
my_path = ...

with my_path.open("w") as f:
    ...
```
which i believe makes this a lot cleaner

Reply:
You can also enable a ruff linter rule for this: https://docs.astral.sh/ruff/rules/#flake8-use-pathlib-pth

They also have explanations as to why they are recommended.

Reply:
Ahh, yes. I had actually forgotten about this piece of code.